api/src/main/java/com/xwiki/azureoauth/internal/AzureADInitializer.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package com.xwiki.azureoauth.internal;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Provider;
+import javax.inject.Singleton;
+
+import org.xwiki.component.annotation.Component;
+import org.xwiki.configuration.ConfigurationSaveException;
+import org.xwiki.extension.ExtensionId;
+import org.xwiki.extension.InstalledExtension;
+import org.xwiki.extension.repository.InstalledExtensionRepository;
+import org.xwiki.model.reference.DocumentReferenceResolver;
+import org.xwiki.query.Query;
+import org.xwiki.query.QueryException;
+import org.xwiki.query.QueryManager;
+import org.xwiki.stability.Unstable;
+import org.xwiki.wiki.descriptor.WikiDescriptorManager;
+
+import com.xpn.xwiki.XWiki;
+import com.xpn.xwiki.XWikiContext;
+import com.xpn.xwiki.XWikiException;
+import com.xpn.xwiki.doc.XWikiDocument;
+import com.xpn.xwiki.objects.BaseObject;
+import com.xwiki.azureoauth.configuration.AzureConfiguration;
+import com.xwiki.azureoauth.configuration.AzureOldConfiguration;
+import com.xwiki.azureoauth.internal.oldConfiguration.OldOAuthAzureConfiguration;
+
+import static com.xwiki.azureoauth.internal.configuration.DefaultAzureConfiguration.OIDC_USER_CLASS;
+
+/**
+ * Handles the initialization of Azure AD with OIDC integration.
+ *
+ * @version $Id$
+ * @since 2.0
+ */
+@Component(roles = AzureADInitializer.class)
+@Singleton
+@Unstable
+public class AzureADInitializer
```

Comment:
After looking at how this class is used, I'll rename it to `EntraIDOIDCMigrator` /  `AzureADOIDCMigrator` since you are mainly using it for migration operations. `Initializer` isn't a very good fit for these